src/modules/products/ProductsSaga.ts

```diff
@@ -85,14 +116,90 @@ function* addToBagSagaHandler(action: addToBagAction) {
     }
 }
 
+/**
+ * Increase Product Count in Order saga
+ */
+function* increaseProductCountSagaHandler(action: increaseProductCountAction) {
+    try {
+        const productId: string | null = R.pathOr(null, ['payload', 'productId'], action);
+        const pieces = yield select(getOrderPiecesSelector);
+        const currentProduct = findProduct(productId, pieces);
+        const currentProductCount: number = R.propOr(0, 'count', currentProduct);
+
+        if (!!productId) {
+            const updatedPieces = increasePieceCount(pieces, productId, currentProductCount);
+
+            yield put({
+                type: successAction(ADD_TO_BAG),
+                payload: updatedPieces,
+            });
+        }
+    } catch (e) {
+        //show error modal
+    }
+}
+
+/**
+ * Decrease Product Count in Order saga
+ */
+function* decreaseProductCountSagaHandler(action: decreaseProductCountAction) {
+    try {
+        const productId: string | null = R.pathOr(null, ['payload', 'productId'], action);
+        const pieces = yield select(getOrderPiecesSelector);
+        const currentProduct = findProduct(productId, pieces);
+        const currentProductCount: number = R.propOr(0, 'count', currentProduct);
+
+        if (currentProductCount <= 1) return;
+
+        if (!!productId) {
```

Comment:
use explicit cast to Boolean

Reply:
I replace` !!productId` =>` !R.isNil(productId).` 
TypeScript wrote error message with string type (because productId null or string and Boolean dont Removes this opportunity) in 
`const updatedPieces = increasePieceCount(pieces, productId, currentProductCount);
`